### PR TITLE
fix(message): preserve whitespace in TOML serialization

### PR DESCRIPTION
**Pull request by TimeToBuildBob**
## Summary

HIGH severity issue from #1035: Silent whitespace stripping during TOML deserialization could corrupt data integrity when whitespace is semantically meaningful (e.g., code indentation).

### Problem
Both `from_toml()` and `toml_to_msgs()` called `.strip()` on message content, silently removing leading/trailing whitespace. This could corrupt:
- Code blocks with intentional indentation
- Formatted text with meaningful whitespace
- Any content where whitespace is semantically significant

### Solution
- Remove `.strip()` from `to_toml()` to preserve whitespace on serialization
- Remove `.strip()` from `from_toml()` and `toml_to_msgs()`
- Add targeted fix to remove only the trailing newline that TOML multiline format adds (the newline before closing `"""`)
- Add test verifying whitespace preservation

### Testing
- All 13 message tests pass
- New `test_toml_preserves_whitespace` test verifies leading whitespace and internal structure are preserved

Related: #1035 (HIGH #2: Silent Whitespace Stripping)

----

> [!IMPORTANT]
> Fixes whitespace stripping issue in TOML serialization by removing `.strip()` calls and adding a function to handle TOML-specific newline artifacts.
> 
>   - **Behavior**:
>     - Remove `.strip()` from `to_toml()`, `from_toml()`, and `toml_to_msgs()` in `message.py` to preserve whitespace.
>     - Add `_fix_toml_content()` to remove trailing newline from TOML multiline strings.
>   - **Testing**:
>     - Add `test_toml_preserves_whitespace` in `test_message.py` to verify whitespace preservation.
>     - All 13 existing message tests pass.
> 
> <sup>This description was created by </sup>[Ellipsis](https://www.ellipsis.dev?ref=gptme%2Fgptme&utm_source=github&utm_medium=referral)<sup> for 6c57f378a2d7f2e5d2851a77ee67aed7ec99cf8d. You can [customize](https://app.ellipsis.dev/gptme/settings/summaries) this summary. It will automatically update as commits are pushed.</sup>

### CODE REVIEW

File: gptme/message.py [hunk 1]
```diff
@@ -252,9 +252,15 @@ def from_toml(cls, toml: str) -> Self:
         if "metadata" in msg and msg["metadata"]:
             metadata = MessageMetadata(**msg["metadata"])
 
+        # Remove exactly one trailing newline that TOML multiline format adds
```
[the remark this answers is not in the document]
Comment (ErikBjare) [reply gptme/message.py]: @TimeToBuildBob address this

**Comment by ErikBjare** [comment]
@TimeToBuildBob address the review comment

**Comment by TimeToBuildBob** [comment]
✅ Addressed the review comment in 6c57f378a

Refactored the duplicated trailing newline removal logic into a shared module-level `_fix_toml_content()` helper function. Both `from_toml()` and `toml_to_msgs()` now use this shared function instead of having duplicate implementations.

File: gptme/message.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (TimeToBuildBob) [reply gptme/message.py]: Done! Extracted `_fix_toml_content()` to module level (commit 6c57f378a) so it's now shared between `from_toml()` and `toml_to_msgs()`.
